File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseVowels = function(s) {
    var arr = s.split("");
    let max = s.length - 1;
    let i = 0;
    var vowels = "aAeEiIoOuU";
    while(i <= max)
    {
        if(vowels.includes(arr[i]) && vowels.includes(arr[max]))
        {
           [arr[i],arr[max]] = [arr[max],arr[i]]
            i++, max--;
        }
        if (!vowels.includes(arr[i])) i++;
        if (!vowels.includes(arr[max])) max--;
    }
    return arr.join("");
};

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cpp
class Solution {
public:
    string reverseVowels(string s) {
        int max = s.size() -1;
        int  i = 0;
        char temp;
        string vowels("aAeEiIoOuU");
        while(i <=  max && max != 0)
        {
            if (vowels.find(s[i]) != -1 && vowels.find(s[max]) != -1)
            {
                temp = s[i];
                s[i++] = s[max];
                s[max--] = temp;
            }
            if(vowels.find(s[i]) == -1) i++;
            if(vowels.find(s[max]) == -1) max--;
        }
        return s;
    }
};